### DISCUSSION TITLE
New Major Release v3.1.0 - "Cleve"

### DISCUSSION THREAD

**Discussion by wahln** [e0404/matRad#792] [Announcements]
### Features
This major release finally combines multiple diverging developments over the last years:

- **File Structure Overhaul** 
- **Scenario Management and robust / 4D optimization**
- **Object-Oriented DoseEngines & new Monte Carlo interfaces** 
- **Helium planning**
- **Extended biological modeling**
- **Widget-Based GUI**
- **DICOM Exporter and updated Importer**
- **Experimental Brachytherapy Module**

Check [CHANGELOG.md](https://github.com/e0404/matRad/blob/v3.1.0/CHANGELOG.md) for more information.

### Standalone
In addition to the Matlab source code, this release also contains updated installers that can provide a standalone version (e.g., for educational purposes). The standalone only features the graphical user interface, and not all new features are accessible. We also have an experimental [Docker Container](https://hub.docker.com/repository/docker/e0404/matrad).

Check our Wiki for how to set-up matRad (as source code or standalone): https://github.com/e0404/matRad/wiki/Setting-up-matRad

### Disclaimer & License
- DO NOT USE MATRAD CLINICALLY -
Check [LICENSE.md](https://github.com/e0404/matRad/blob/v3.1.0/LICENSE.md) and [README.md](https://github.com/e0404/matRad/blob/v3.1.0/README.md) for more infos

### Funding
This release has been supported through the following research grants:
-   The German Research Foundation (DFG), Project No. 265744405 & 443188743
-   The German Cancer Aid, Project No. 70113094
-   The German Federal Ministry of Education and Research (BMBF), Project No. 01DN17048
-   Mathworks Academic Research Support

**Comment by wahln** [comment]
**After the release is before the release!**

Feel free to post any comments regarding the new features or what you would wish to see in the future!

**Reply by Yoonho-Na** [reply]
@wahln 
mac installer doesn't work.
It says that the file is damaged.
Of course I did allow security thing in the mac settings.

and I found out that matrad v2.10.1 installs well.

mac os : Sequoia 15.1(24B83)

**Reply by wahln** [reply]
Thanks for checking the Mac Installer on your system - I will investigate/rebuild and get back to you. Unfortunately we don't have that many Mac-Systems to test.

**Reply by wahln** [reply]
I think the issue still has to do with that the matRad App is not Apple Developer verified. It seems like dmgs with unsigned Apps are blocked and reported as damaged on newer systems. I need to figure out if I can find another way to package it.

**Reply by wahln** [reply]
One more update - it seems like on newer MacOS unverified files are automatically quarantined. 
Steps I just did to get the Downloaded installer to run:
1. Copy the Installer from the Image to somewhere on your system (e.g. Desktop)
2. Run `sudo xattr -rd com.apple.quarantine /Path/To/Installer/matRad_installerMac64_v3.1.0.app` on the file. `sudo` is required. Afterwards the installer starts

At the end of the installer, the installer notifies you about environment variables that need to be set, pointing to the Matlab Runtime that is installed during installation. So keep track of the Runtime path and set the environment variable later. If System Integrity Protection is enabled, you might be required to write a small "startup" script for matRad to launch.

Sorry that this is such a hassle, but Apple goes a long way to only launch applications that are verified (which means, for developers, to pay money to them).

**Reply by wahln** [reply]
If the installation worked, and assuming you used standard paths, here's a Terminal command that's launching the App correctly for me with enabled System Integrity Protection:
`/Applications/matRad/application/run_matRad.sh /Applications/MATLAB/MATLAB_Runtime/R2024b/`

**Reply by Yoonho-Na** [reply]
@wahln thank you I successfully installed the new MatRad.

I have few requests.

First,

> Run sudo xattr -rd com.apple.quarantine /Path/To/Installer/matRad_installerMac64_v3.1.0.app on the file. sudo is required. Afterwards the installer starts

this is confusing because the the matRad_installerMac64_v3.1.0.app doesn't show up until I execute the .dmg file.
So I used the `Run sudo xattr -rd com.apple.quarantine /Path/To/Installer.dmg/matRad_installerMac64_v3.1.0.dmg` instead.
This still works I guess(.app -> .dmg) but I think there should be more detailed explanation.

<img width="323" alt="스크린샷 2024-11-26 오전 9 28 28" src="https://github.com/user-attachments/assets/d58f58b8-1292-4c84-82b3-2f035814e89e">


Second,

`/Applications/matRad/application/run_matRad.sh /Applications/MATLAB/MATLAB_Runtime/R2024b/`

The last '/' seems inappropriate because I think inside of the shell script, it just adds up two directory, so It results in two '//' like below.

`DYLD_LIBRARY_PATH is .:/Applications/MATLAB/MATLAB_Runtime/R2024b//runtime/maci64:/Applications/MATLAB/MATLAB_Runtime/R2024b//bin/maci64:/Applications/MATLAB/MATLAB_Runtime/R2024b//sys/os/maci64`

And Lastly,

The MatRad crashes without the the run script.
I always have to use terminal to run MatRad.
This is little annoying so I hope this is fixed near future.

****
EDIT:
One more thing.
DICOM export and import doesn't work.
even though the RT struct and RT CT slice seems exported, It cannot be opened in the MatRAd.
and RT Dose is not exported because of the unknown types.

<img width="510" alt="스크린샷 2024-11-26 오전 10 44 20" src="https://github.com/user-attachments/assets/882d6501-463a-442e-be0d-f4363ab38bca">
<img width="534" alt="스크린샷 2024-11-26 오전 10 44 02" src="https://github.com/user-attachments/assets/3c12581c-2fa9-4a50-b51c-73f76859f221">

**Reply by wahln** [reply]
Regarding the app v. dmg, that's why my first step was to copy the installer app out of the dmg image to the desktop:
> 1. Copy the Installer from the Image to somewhere on your system (e.g. Desktop)
But I am glad this works also directly on the dmg!

Regarding the slash in the path, that shouldn't do any damage.

Regarding the Problem with the need to set up a run script. Currently I do not see what we can do about it. It's a combined issue with the MATLAB Compiler and the way it produces standalone apps and Mac's System Integrity Protection hating every non-certified app. 
If we or someone else figures out a better way, we will put it in here.
One option I see: 
1. Put the `/Applications/matRad/application/run_matRad.sh /Applications/MATLAB/MATLAB_Runtime/R2024b` command in your own script, e.g. `matRad_launcher.sh`
2. Put the `matRad_launcher.sh` on the Desktop.
3. Using the "Open with..." Dialog to set the script to be automatically opened with Utilities->Terminal. 

Now, regarding the DICOM export.
It's true that we currently do only export physicalDose and RBExDose. This might not be ideal, I agree. As a workaround, you could export the other quantities to binary format using the binary exporter.
And regarding the import:
There seems to be a bug when you try to import RTDose at the moment without an RTPlan / if the dose does not have `RTPlanReferenceSequence`. Will open an issue for that.

Finally, thanks so much for the detailed feedback!

**Reply by wahln** [reply]
DICOM issues are worked on in #797 and #798

**Reply by Yoonho-Na** [reply]
@wahln 
Thank you for the explanation.
I want to note that the DICOM Import also doesn't work.
![image](https://github.com/user-attachments/assets/90bc277a-55b4-4f31-a281-94ee50ef3c09)
This screenshot is showing the directory for DICOM import.
but non of them could be opened 
(grey-ish font color means its disabled. it should be white colored text if it's able to open)

Does DICOM import work well on windows??

**Reply by wahln** [reply]
You do not select files in the dicom import (that's why they are greyed out), but you select the whole folder. The DICOM import will then scan the folder and show you the series IDs and files that are connected and can be imported.

**Reply by Yoonho-Na** [reply]
Oh I see.
I deleted all the RT dose file above (only ct and struct remains) and tried to import it.

![스크린샷 2024-11-26 오후 8 43 31](https://github.com/user-attachments/assets/787897d8-dbca-40c6-a82f-9ff4b6005df1)

After pressing import button on the left side. why is this suddenly saving into mat file??
![스크린샷 2024-11-26 오후 8 44 29](https://github.com/user-attachments/assets/fbcf9b70-b1b1-4231-a1ca-b56edb328011)

I couldn't understand but I did save it.
after saving mat file, CT and Struct is still not imported on the interface.
and as I keep click the import button again, there's error message on the terminal.

![스크린샷 2024-11-26 오후 8 52 49](https://github.com/user-attachments/assets/385e5e7f-9526-4d1f-b0d4-ef82b9705532)
Sorry for the korean character.

the below error message means

```
This type of variable cannot be indexed using curly braces.
Error in: matRad_importDicomWidget/import_button_Callback (line 177)
Error in: matRad_importDicomWidget>@(hObject,event)import_button_Callback(this,hObject,event) (line 714)
An error occurred while executing the UIControl Callback.
```

**Reply by Yoonho-Na** [reply]
@wahln Oh nevermind the above issue.
dicom import worked by simply just pressing refresh button.

**Comment by AndrewFielding** [comment]
Trying to write out the plan information to TOPAS output files using the latest matRad v3.. Getting the following error:

Warning: Using default stf generator Particle IMPT stf Generator! 
Forward dose calculation using  'TOPAS' Dose Engine...
Dose grid has dimensions 512x512x135
Adjusting structures for overlap... Done!
Resampling structure set... Done!
Adjusting structures for overlap... Done!
Resampling cst... Unrecognized field name "cube".

CT data and structures were imported. There is no ct.cube array in the ct structure only ct.cubeHU. I think for proton calculations ct.cube should contain stopping powers for each voxel, from the technical documentation. I noticed that the example datasets/plans have a ct.cube data array.

**Reply by wahln** [reply]
This should be fixed now on the latest dev branch and will be in the next release.
